lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -318,11 +312,12 @@ void onStart() {
 
     try {
 
+      String[] url = this.trackingUrl.split(":");
       LOG.log(Level.FINE, "YARN registration: register AM at \"{0}:{1}\" tracking URL \"{2}\"",
-          new Object[] {AM_REGISTRATION_HOST, AM_REGISTRATION_PORT, this.trackingUrl});
+          new Object[] {url[0], Integer.parseInt(url[1]), this.trackingUrl});
 
       this.registration.setRegistration(this.resourceManager.registerApplicationMaster(
-          AM_REGISTRATION_HOST, AM_REGISTRATION_PORT, this.trackingUrl));
+          url[0], Integer.parseInt(url[1]), this.trackingUrl));
```

Comment:
>url[0], Integer.parseInt(url[1] [](start = 10, length = 31)

this hardcoded parsing seem dangerous (what if trackingUrlProvider changes behavior?), 
otherwise looks ok to me.